lib/srv/reexec.go

```diff
@@ -233,6 +246,62 @@ func RunCommand() (io.Writer, int, error) {
 		return errorWriter, teleport.RemoteCommandFailure, trace.Wrap(err)
 	}
 
+	if c.X11Config != nil {
+		// Open x11rdy fd to signal parent process once X11 forwarding is set up.
+		x11rdyfd := os.NewFile(uintptr(5), "/proc/self/fd/5")
```

Comment:
I see this is closed at the end of the block, but should you defer the close instead?

If an error occurs, this never gets closed - is that what you intend?

In fact, it looks like you don't write to `x11rdyfd` until the very end - can you move this code down closer to where it's needed so that you never even open it if an error is encountered?